develop/dev-guide-sample-application-nextjs.md

```diff
@@ -0,0 +1,265 @@
+
+3. If you have not set a password yet, click **Create password** to generate a random password.
+
+   <Tip>
+   
+   If you have generated a password before, you can use the original password directly, or click **Reset password** to generate a new password.
+   
+   </Tip>
+
+4. Copy and rename `.env.example` to `.env`:
+
+   ```bash
+   # Linux
+   cp .env.example .env
+   ```
+
+   ```powershell
+   # Windows
+   Copy-Item ".env.example" -Destination ".env"
+   ```
+
+5. Copy and paste the corresponding connection string into `.env`. The following is an example:
+
+   ```bash
+   TIDB_HOST='{gateway-region}.aws.tidbcloud.com'
+   TIDB_PORT='4000'
+   TIDB_USER='{prefix}.root'
+   TIDB_PASSWORD='{password}'
+   TIDB_DB_NAME='test'
+   ```
+
+   Replace the placeholders in `{}` with the values obtained in the **Connect** window.
+
+6. Save the file.
+
+</div>
+
+<div label="TiDB Self-Hosted">
+
+1. Copy and rename `.env.example` to `.env`:
+
+   ```bash
+   # Linux
+   cp .env.example .env
+   ```
+
+   ```powershell
+   # Windows
+   Copy-Item ".env.example" -Destination ".env"
+   ```
+
+2. Copy and paste the corresponding connection string into `.env`. The following is an example:
+
+   ```bash
+   TIDB_HOST='{tidb_server_host}'
+   TIDB_PORT='4000'
+   TIDB_USER='root'
+   TIDB_PASSWORD='{password}'
+   TIDB_DB_NAME='test'
+   ```
+
+   Replace the placeholders in `{}` with the values obtained in the **Connect** window. If you are running TiDB on your local machine, the default Host address is `127.0.0.1` and the password is empty.
+
+3. Save the file.
+
+</div>
+
+</SimpleTab>
+
+### Step 4: Run the sample application locally
+
+1. Start the application:
+
+   ```bash
+   npm run dev
+   ```
+
+2. Open your browser and visit `http://localhost:3000`. (Check your terminal for the actual port number, and the default is `3000`.)
+
+3. Click the `run SQL` button to execute the sample code.
+
+4. Check the output in the terminal. If the output is similar to the following, the connection is successful:
+
+   ```json
+   {
+     "results": [
+       {
+         "Hello World": "Hello World"
+       }
+     ]
+   }
+   ```
+
+## Sample code snippets
+
+You can refer to the following sample code snippets to complete your own application development.
+
+For complete sample code and how to run it, check out the [tidb-nextjs-vercel-quickstart](https://github.com/tidb-samples/tidb-nextjs-vercel-quickstart) repository.
+
+### Connect to TiDB
+
+```javascript
+/** It's different from the sample code in the repository, which is for learning purposes only.
+ * Which is not recommended in production environment.
+ * Please refer to the code in [tidb-nextjs-vercel-quickstart](https://github.com/tidb-samples/tidb-nextjs-vercel-quickstart) GitHub repository for production environment.
+ */
+import mysql from 'mysql2';
+
+pool = mysql.createPool({
```

Comment:
```suggestion
// lib/tidb.js
import mysql from 'mysql2';

const pool = mysql.createPool({
```

Put the code for establishing the connection in a separate file, like `lib/tidb.js`.

For example: https://github.com/steven-tey/dub/blob/main/lib/planetscale.ts